tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -298,7 +298,7 @@ def result_from_exception(self, test, e):
         message = text_type(getattr(e, "message", ""))
         if message:
             message += "\n"
-        message += traceback.format_exc(e)
+        message += traceback.format_exc()
```

Comment:
I'm not clear what change you're requesting here @jgraham . I agree with your statement, but this PR doesn't change whether or not `traceback.format_exc()` prints `e` or the previous exception, so if this was buggy before, it still is.

Looking at the code, `result_from_exception` is only called from line 259 I believe, so it is currently correct, if misleading.

Reply:
That might be true, but we should definitely fix this whilst we're looking at the code anyway because it's such an obvious hazard and so likely to be unfixed if we don't do it now.

Unfortunately Python 2 makes this hard to do correctly. I think this function needs to take a traceback object, which we can collect at the point of the exception being thrown like `_, _, tb = sys.exc_info()` and then when formatting the exception do something like `"".join(traceback.format_exception(type(e), e, tb))` to construct the actual string.

Reply:
Fixing it now seems a reasonable request. Could we not just have the function take the result of `traceback.format_exc()` as an argument?

```
        except Exception as e:
            exception_string = traceback.format_exc()
            self.logger.warning(exception_string)
            result = self.result_from_exception(test, e, exception_string)
```

Reply:
That would also work.

Reply:
Updated the code.  Thank you!